src/common/iso_8601.cc

```diff
@@ -41,7 +41,8 @@ optional<real_time> calculate(const tm& t, uint32_t n = 0) {
     return none;
   }
 
-  return real_clock::from_time_t(tt) + nanoseconds(n);
+  return boost::make_optional<real_time>(real_clock::from_time_t(tt)
+                                         + nanoseconds(n));
 }
```

Comment:
This looks fine. If it works I'd rather use something like
```c++
  return { real_clock::from_time_t(tt) + nanoseconds(n) };
```

Since that will continue to work if we switch to std::optional, but it's by no means necessary.